homeassistant/components/scene/__init__.py

```diff
@@ -70,7 +69,7 @@ def _platform_validator(config):
 
     hass.services.async_register(
```

Comment:
Not entity register.

Reply:
Want to make sure I understand – `ENTITY_SERVICE_SCHEMA` should only be used in `async_register_entity_service` calls (and not, as I had been doing here, anytime a service schema used entity IDs)?

Reply:
_It depends_ 😄 

So the support for area_id is part of `EntityComponent.async_extract_from_service` and `helpers/service.py#async_extract_entity_ids` functions. So if a service handler uses that, it is fine. `async_register_entity_service` does this. For others you need to look at the specific service implementation.

Reply:
Okay, I'll dig in. 😫😆

Reply:
This PR can be done piecemeal and incremental. Just focus on `async_register_entity_service` initially.

Reply:
I was thinking that perhaps we should offer a service registration API equivalent to `async_register_entity_service` that platforms can use that does all the entity id resolution logic so we don't have various forms of implementations and instead sends in the resolved IDs to the service? Not related to this PR, but if you think it would be a good idea, I can work on that and update the other platforms.

Reply:
We have a helper function `EntityComponent.async_extract_from_service`, it's just not available for platforms, but would help if it was.